presidio-image-redactor/tests/test_dicom_image_pii_verify_engine.py

```diff
@@ -0,0 +1,929 @@
+    Args:
+        ocr_engine (TesseractOCR): Tesseract OCR engine or None.
+        image_analyzer_engine (ImageAnalyzerEngine): Presidio image analyzer engine or None.
+    """
+    try:
+        test_engine = DicomImagePiiVerifyEngine(ocr_engine, image_analyzer_engine)
+    except:
+        raise TypeError("Invalid input into initializing")
+
+
+# ------------------------------------------------------
+# DicomImagePiiVerifyEngine.verify_dicom_instance()
+# ------------------------------------------------------
+def test_verify_dicom_instance_happy_path(
+    mocker,
+    mock_engine: DicomImagePiiVerifyEngine,
+    mock_instance: pydicom.dataset.FileDataset,
+):
+    """Test happy path for DicomImagePiiVerifyEngine.verify_dicom_instance
+    Args:
+        mock_engine (DicomImagePiiVerifyEngine): Instantiated engine.
+        mock_instance (pydicom.dataset.FileDataset): Loaded DICOM.
+    """
+    # Assign
+    padding_width = 25
+
+    mock_greyscale = mocker.patch.object(
+        DicomImagePiiVerifyEngine, "_check_if_greyscale", return_value=None
+    )
+    mock_rescale_array = mocker.patch.object(
+        DicomImagePiiVerifyEngine, "_rescale_dcm_pixel_array", return_value=None
+    )
+    mock_save_pixel_array = mocker.patch.object(
+        DicomImagePiiVerifyEngine, "_save_pixel_array_as_png", return_value=None
+    )
+    mock_image_open = mocker.patch(
+        "presidio_image_redactor.dicom_image_pii_verify_engine.Image.open",
+        return_value=None,
+    )
+    mock_add_padding = mocker.patch.object(
+        DicomImagePiiVerifyEngine, "_add_padding", return_value=None
+    )
+    mock_get_metadata = mocker.patch.object(
+        DicomImagePiiVerifyEngine, "_get_text_metadata", return_value=[None, None, None]
+    )
+    mock_make_phi_list = mocker.patch.object(
+        DicomImagePiiVerifyEngine, "_make_phi_list", return_value=None
+    )
+    mock_patternrecognizer = mocker.patch(
+        "presidio_image_redactor.dicom_image_pii_verify_engine.PatternRecognizer",
+        return_value=None,
+    )
+    mock_perform_ocr = mocker.patch.object(
+        TesseractOCR, "perform_ocr", return_value=None
+    )
+    mock_analyze = mocker.patch.object(
+        ImageAnalyzerEngine, "analyze", return_value=None
+    )
+    mock_verify = mocker.patch.object(
+        DicomImagePiiVerifyEngine, "verify", return_value=None
+    )
+
+    # Act
+    _, _, _ = mock_engine.verify_dicom_instance(mock_instance, padding_width)
+
+    # Assert
+    assert mock_greyscale.call_count == 1
+    assert mock_rescale_array.call_count == 1
+    assert mock_save_pixel_array.call_count == 1
+    assert mock_image_open.call_count == 1
+    assert mock_add_padding.call_count == 1
+    assert mock_get_metadata.call_count == 1
+    assert mock_make_phi_list.call_count == 1
+    assert mock_patternrecognizer.call_count == 1
+    assert mock_perform_ocr.call_count == 1
+    assert mock_analyze.call_count == 1
+    assert mock_verify.call_count == 1
```

Comment:
What is it testing? and how will you enforce future addition of new method calls? Worth adding that one is called with the others return value?

Reply:
To be honest, this unit test does not test much given that `.verify_dicom_instance()` mostly consists of just calling other functions. I made this to help with code coverage.

The true testing of the functionality of `.verify_dicom_instance()` as a whole occurs in the integration test.